source/common/config/incremental_subscription_factory.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
```

Comment:
Why can't the regular `SubscriptionFactory` handle incremental xDS as an additional case?

Reply:
Because [Incremental]SubscriptionCallbacks are two different types, so that they can have different argument types. I could merge them and add onIncrementalConfigUpdate (or an overload of onConfigUpdate), and have implementations always leave one or the other with a "not implemented" stub implementation.

Reply:
(talked offline; going to do that for now, and then in a follow-up translate all xDSes' usage of onConfigUpdate to the incremental style one)